### DISCUSSION TITLE
There is no response from the websocket server after a certain amount of time

### DISCUSSION THREAD

**Discussion by jbc0708** [fastapi/fastapi#6001] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# FastAPI WebSocket Server Code
from typing import List, Dict
from fastapi import FastAPI, Request, WebSocket, WebSocketDisconnect
from fastapi.responses import HTMLResponse
from fastapi.staticfiles import StaticFiles
from fastapi.templating import Jinja2Templates
import datetime
import uvicorn
import json

app = FastAPI()

templates = Jinja2Templates(directory="templates")
app.mount("/static", StaticFiles(directory="static"), name="static")

class ConnectionManager:
    def __init__(self):
        self.rooms = {}
        #self.active_connections: List[WebSocket] = []

    async def connect(self, room: str, websocket: WebSocket):
        await websocket.accept()
        if room not in self.rooms: 
            self.rooms[room] = []
        self.rooms[room].append(websocket)
        
    def disconnect(self, room: str, websocket: WebSocket):
        self.rooms[room].remove(websocket)
        if len(self.rooms[room]) == 0: del self.rooms[room]

    async def send_personal_message(self, message: str, websocket: WebSocket):
        await websocket.send_text(message)

    async def broadcast(self, room: str, message: str):
        for connection in self.rooms[room]:
            await connection.send_text(message)

manager = ConnectionManager()

@app.get("/", response_class=HTMLResponse)
async def index(request: Request):
    content = {
        "request": request,
        "currentTime": datetime.datetime.now().__str__().split(".")[0],
        "currentState": "Testing",
    }
    return templates.TemplateResponse("index.html", content)

@app.websocket("/ws/{room}/{client_id}")
async def websocket_endpoint(websocket: WebSocket, room: str, client_id: str):
    print("Room: %s ID: %s Connected" % (room, client_id))
    await manager.connect(room, websocket)
    try:
        while True:
            data = await websocket.receive_text()
            print(data)
            await manager.broadcast(room, data)
    except WebSocketDisconnect:
        manager.disconnect(room, websocket)
        #await manager.broadcast(room, f"Client #{client_id} left the chat")

if __name__ == "__main__":
    uvicorn.run(app, host="127.0.0.1", port=9999 ,ws_ping_interval=36000.0, ws_ping_timeout=36000.0)

# Client: Message Reciever Code
import websocket
import json
import random
import time
import datetime
import sys, os
import sqlite3

if __name__ == "__main__":
    ws = websocket.WebSocket()
    uri = 'ws://127.0.0.1:9999/ws/1/1'
    ws.connect(uri)

    while True:
        temp = ws.recv()
        now = datetime.datetime.now().__str__()
        print(now, temp)
    ws.close()

# Client: Message Sender Code
import websocket
import json
import random
import time
import datetime
import sys

def connectServer(target, room, id):
    uri = 'ws://127.0.0.1:9999/ws/1/2
    target.connect(uri)

if __name__ == "__main__":
    ws = websocket.WebSocket()
    connectServer(ws, room, id)

    while True:
        data = "%s number: %s" % (datetime.datetime.now().__str__().split(".")[0], random.randint(1000,9999))
        ws.send(data)
        print("Send: ",data)
        time.sleep(0.01)
        
    ws.close()
```

### Description

I created a websocket server and 2 clients using FastAPI and websocket-client. And I ran a test to see if data transmission/reception was normal.
Both the server and the client were connected normally. Data was sent from client 1 to client 2 at intervals of 0.01 seconds. After about 5 minutes, client 1 continued to send data, but client 2 did not receive any data.
I felt strange, and all of the client 1, client 2, and server print moving messages.
When the problem occurred, I confirmed that client 1 was sending messages but not reaching the server. 
The server did not generate any error messages for Client1.
Why is this problem occurring and is there a solution?

### Operating System

macOS

### Operating System Details

MacBook M1 Air 16G. Ventura 13.1

### FastAPI Version

0.88.0

### Python Version

3.11

**Comment by Kludex** [comment]
The `ws_ping_interval` should be the issue... Why did you set it so high?

**Reply by jbc0708** [reply]
Becaouse if ws_ping_interval isn't set,   clients are all disconnected after 20 seconds later... so i set 2 options(ws_ping_interval,ws_ping_timeout)..

**Reply by Kludex** [reply]
I think if you set those properly you are able to notice when they are disconnected, and if not, they are disconnected and you have the behavior you are having. :thinking:

**Reply by jbc0708** [reply]
I remove that option(ws_ping_interval) and tried again, but the problem still occurs
The websocket-client module seemed to have a problem, so I replaced it with the websockets module and proceeded asynchronously.
but issue still happen again....
